src/net/p2p.rs

```diff
@@ -3294,12 +3295,17 @@ impl PeerNetwork {
         &mut self,
         url: &UrlString,
         addr: &SocketAddr,
+        sortdb: &SortitionDB,
         mempool: &MemPoolDB,
         chainstate: &mut StacksChainState,
     ) -> Result<(bool, Option<usize>), net_error> {
+        let burn_tip = SortitionDB::get_canonical_burn_chain_tip(sortdb.conn())?;
```

Comment:
Instead of pulling through the `SortitionDB` and doing additional database I/O on this hot path, can you instead use the cached chain tip in `self.burnchain_tip`?